oracle/src/oracleimporter.cpp

```diff
@@ -76,33 +76,38 @@ CardInfo *OracleImporter::addCard(const QString &setName,
     cardName = cardName.replace("Æ", "AE");
     cardName = cardName.replace("’", "'");
 
-    // Remove {} around mana costs
-    cardCost.remove(QChar('{'));
-    cardCost.remove(QChar('}'));
-
-    // detect mana generator artifacts
-    bool mArtifact = false;
-    if (cardType.endsWith("Artifact"))
-        for (int i = 0; i < cardTextRows.size(); ++i)
-            if (cardTextRows[i].contains("{T}") && cardTextRows[i].contains("to your mana pool"))
-                mArtifact = true;
-
-    // detect cards that enter the field tapped
-    bool cipt = cardText.contains("Hideaway") || (cardText.contains(cardName + " enters the battlefield tapped") && !cardText.contains(cardName + " enters the battlefield tapped unless"));
-    
-    // insert the card and its properties
-    CardInfo *card = new CardInfo(this, cardName, isToken, cardCost, cmc, cardType, cardPT, cardText, colors, relatedCards, upsideDown, cardLoyalty, cipt);
-    int tableRow = 1;
-    QString mainCardType = card->getMainCardType();
-    if ((mainCardType == "Land") || mArtifact)
-        tableRow = 0;
-    else if ((mainCardType == "Sorcery") || (mainCardType == "Instant"))
-        tableRow = 3;
-    else if (mainCardType == "Creature")
-        tableRow = 2;
-    card->setTableRow(tableRow);
-    
-    cards.insert(cardName, card);
+    CardInfo * card;
+    if (cards.contains(cardName)) {
+        card = cards.value(cardName);
+    } else {
+        // Remove {} around mana costs
+        cardCost.remove(QChar('{'));
+        cardCost.remove(QChar('}'));
+
+        // detect mana generator artifacts
+        bool mArtifact = false;
+        if (cardType.endsWith("Artifact"))
+            for (int i = 0; i < cardTextRows.size(); ++i)
+                if (cardTextRows[i].contains("{T}") && cardTextRows[i].contains("to your mana pool"))
+                    mArtifact = true;
+
+        // detect cards that enter the field tapped
+        bool cipt = cardText.contains("Hideaway") || (cardText.contains(cardName + " enters the battlefield tapped") && !cardText.contains(cardName + " enters the battlefield tapped unless"));
```

Comment:
I didn't think so, just pointing out a potential for later. :+1:

On Thu, Jul 2, 2015 at 12:02 PM ctrlaltca <notifications@github.com> wrote:

> In oracle/src/oracleimporter.cpp
> <https://github.com/Cockatrice/Cockatrice/pull/1190#discussion_r33793312>:
>
> > +    if (cards.contains(cardName)) {
> > +        card = cards.value(cardName);
> > +    } else {
> > +        // Remove {} around mana costs
> > +        cardCost.remove(QChar('{'));
> > +        cardCost.remove(QChar('}'));
> > +
> > +        // detect mana generator artifacts
> > +        bool mArtifact = false;
> > +        if (cardType.endsWith("Artifact"))
> > +            for (int i = 0; i < cardTextRows.size(); ++i)
> > +                if (cardTextRows[i].contains("{T}") && cardTextRows[i].contains("to your mana pool"))
> > +                    mArtifact = true;
> > +
> > +        // detect cards that enter the field tapped
> > +        bool cipt = cardText.contains("Hideaway") || (cardText.contains(cardName + " enters the battlefield tapped") && !cardText.contains(cardName + " enters the battlefield tapped unless"));
>
> There are currently 632 cards with containing the "enters the battlefield
> tapped" text, most of them are duplicates so we end up with only 259 unique
> cards containing that string.
> None of them contains a shortened version of the card name. Some of them
> contains other text (eg. hideaway lands use "this land" instead of the card
> name). The longest name i found is "Valakut, the Molten Pinnacle"
> So there's no problem here, at least until now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cockatrice/Cockatrice/pull/1190/files#r33793312>.
>